pravega/pravega-operator
# Issue 430: Exposing Segmentstore Service on single IP

### Change log description

Exposing Segmentstore Service on single IP address and Different ports

### Purpose of the change
fixes #430 

### What the code does
Currently, Segment store pods are exposed using external IP address which is given by Service, Each pod has its own service which has a unique IP address which can be used to communicate to the Segment store, after this change we can expose segment store services on the same I/P address and different port

### How to verify it
1) To verify this change we need Bare metal kubernetes cluster and we need to specify the below fields for it to work in 
   the manifest 
```
pravega:
    . . .
    segmentStoreLoadBalancerIP: "10.243.39.103"
    segmentStoreExternalTrafficPolicy: "cluster"
    segmentStoreSvcAnnotations:
      metallb.universe.tf/allow-shared-ip: "shared-ss-ip"
```
and after deploying pravega we should see that all the Segment Store services should be having same LoadBalancerIP and a different port.

2) In case of an environment which does not allow metallb service to provision the same LoadBalancerIP to services like PKS we should be able to run segment store services on different I/P addresses and same port.

I have verified both Scenarios work
